containers/unit_tests/TestScatterView.hpp

```diff
@@ -411,6 +458,88 @@ struct test_scatter_view_impl_cls<DeviceType, Layout, Duplication, Contribution,
       EXPECT_TRUE(std::fabs((val2 - 4.0) / 4.0) < 1e-14);
     }
   }
+
+  // check for correct padding
+  void validateResultsForSubview(
+      orig_view_type orig, std::pair<size_type, size_type>& subRangeDim0,
+      std::pair<size_type, size_type>& subRangeDim1) {
+    (void)subRangeDim1;
+    auto host_view =
+        Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), orig);
+    Kokkos::fence();
+    for (size_type i = 0; i < host_view.extent(0); ++i) {
+      auto val0 = host_view(i, 0);
+      auto val1 = host_view(i, 1);
+      auto val2 = host_view(i, 2);
+      if (i >= std::get<0>(subRangeDim0) && i < std::get<1>(subRangeDim0)) {
+        // is in subview
+        EXPECT_TRUE(std::fabs((val0 - 16.0) / 16.0) < 1e-14);
+        EXPECT_TRUE(std::fabs((val1 - 8.0) / 8.0) < 1e-14);
+        EXPECT_TRUE(std::fabs((val2 - 4.0) / 4.0) < 1e-14);
+      } else {
+        // is outside of subview
+        EXPECT_NEAR(val0, NumberType(0), 1e-14);
+        EXPECT_NEAR(val1, NumberType(0), 1e-14);
+        EXPECT_NEAR(val2, NumberType(0), 1e-14);
+      }
+    }
+  }
+};
+
+// Test ScatterView on subview
+template <typename DeviceType, typename Layout, typename Op,
+          typename NumberType>
+struct test_default_scatter_sub_view {
+ public:
+  using default_duplication = Kokkos::Impl::Experimental::DefaultDuplication<
+      typename DeviceType::execution_space>;
+  using Duplication  = typename default_duplication::type;
+  using Contribution = typename Kokkos::Impl::Experimental::DefaultContribution<
+      typename DeviceType::execution_space, Duplication>::type;
+  using scatter_view_def =
+      typename test_scatter_view_impl_cls<DeviceType, Layout, Duplication,
+                                          Contribution, Op,
+                                          NumberType>::scatter_view_type;
+  using orig_view_def =
+      typename test_scatter_view_impl_cls<DeviceType, Layout, Duplication,
+                                          Contribution, Op,
+                                          NumberType>::orig_view_type;
+
+  using size_type = typename Kokkos::HostSpace::size_type;
+
+  void run_test(int n) {
+    // Test creation via create_scatter_view overload 1
+    {
+      orig_view_def original_view("original_view", n);
+
+      auto rangeDim0 = std::pair<size_type, size_type>(0 + 1, n - 1);
+      auto rangeDim1 =
+          std::pair<size_type, size_type>(0, original_view.extent(1));
+
+      auto original_sub_view =
+          orig_view_def(original_view, rangeDim0, rangeDim1);
```

Comment:
is `orig_view_def` the right type here? why not just do `Kokkos::subview(original_view, rangeDim0, rangeDim1)`?

Reply:
Yes. Both, calling Kokkos::subview or the View constructor overload that accepts ranges should give a subview of type auto.

Reply:
orig_view_def constructor creates a view of type orig_view_def. It will never do any auto thing.

Reply:
Ah yes, but with the correct ViewMapping which includes the ViewOffsets. Do you think there is a difference?

Reply:
Yes there is a difference because in this call `original_sub_view` must be of type `orig_view_def`. So if you were to write this:

```c++
      auto rangeDim0 = std::pair<size_type, size_type>(0 + 1, n - 1);
      auto rangeDim1 =
          std::pair<size_type, size_type>(0, original_view.extent(1)-1);

      auto original_sub_view =
          orig_view_def(original_view, rangeDim0, rangeDim1);
```
It wouldn't work because you would try to generate a subview with a second extent of 11, but the type says the second extent is compile time 12.

Reply:
Gotcha, thx! I'll update.